### ISSUE TITLE
[REQUEST] Ability to add system rule support

### ISSUE THREAD

**Issue by dovrosenberg** [dovrosenberg/foundry-simple-weather#27]
**Describe the solution you'd like**
It would be helpful in certain systems if the weather could be tied to relevant rules, to automatically bring up rules when the associated weather comes up.  

**Additional context**
Perhaps something like: https://github.com/orangetruth/weather-effects-5e but it should be designed in a modular way (likely using an API) so that we can support multiple systems and 3rd parties can add support for other systems without needing to bloat this package.

**Comment by dovrosenberg** [comment]
Current thinking is to open an api that easily allows another module to do the actual weather selection, while using the season/climate/biome info and FX generation of this module.  Then for different systems (or just custom rules/roll tables), you'd just install this package plus the other one.  This would also allow those add-on packages to create other effects (like damage or other modifiers to characters).

Will probably implement one (maybe https://github.com/ntriel/pf1-weather-macros, which is pretty simple) as an example.  

The API would need to allow the temperature and the specific weather type to be set for FX purposes, but also a separate text description of the weather could be optionally used to replace the default.  That way you could say "light clouds" but use the FX for "fluffy clouds" or whatever.  The temperature could also be specified to generate randomly around a value to add a little variety without complicating the add-on.

**Comment by SwedishRabbit** [comment]
I have not looked into the latest releases of this module, but is it possible to attach each weather with a 'rule note "? For instance when it is heavy rain a rule window say"...-2 for Lead The Way "

**Comment by dovrosenberg** [comment]
It doesn't do that today, but I think that could be the first version of this enhancement.  It would require someone to put in the roles for ~325 weather conditions, which is fine if one were building a module, but may be overwhelming for someone just trying to put in a couple homebrew notes.  Are you wanting custom rules or rules tied to a specific system?  If custom rules, would you be open to having to put the notes in across 325 weather conditions (or at least the ones where you want the notes to show?)

**Comment by SwedishRabbit** [comment]
> It doesn't do that today, but I think that could be the first version of this enhancement.  It would require someone to put in the roles for ~325 weather conditions, which is fine if one were building a module, but may be overwhelming for someone just trying to put in a couple homebrew notes.  Are you wanting custom rules or rules tied to a specific system?  If custom rules, would you be open to having to put the notes in across 325 weather conditions (or at least the ones where you want the notes to show?)

Well i think the way i would like it as a user is like this:

 I am a GM, i prep my world. I go into the Weather app and select those weathers i think are extreme. I enter some text into the attached "rule effect window " for each and every one. Maybe i could multiselect, and write comments in one go.

 I wouldnt mind spending some time with this as a GM prepping my world. I think other GMs also would be fine with that. And in the long run, maybe GMs could save rulesets and even include them in future updates. But GMs should always decide if to load a ruleset or if to make your own.

**Comment by SwedishRabbit** [comment]
... Continuing... Then the rulesets could be named after which Role playing game they support

**Comment by SwedishRabbit** [comment]
And also it would be really nice if that rule-effect would be sent to the chat below the weather description. I really hope it could be done 🙂

**Comment by dovrosenberg** [comment]
Yes.. I think this is what I'll take on next. Just working through how to best present and store the data.

**Comment by SwedishRabbit** [comment]
> Yes.. I think this is what I'll take on next. Just working through how to best present and store the data.

Wow, sounds great! Looking forward to it 🙂🙂

**Comment by dovrosenberg** [comment]
The initial version of this feature has been released in v1.12.0.  Will continue to evaluate other options, including:
* Ability to save/load from files so that you can have different text for different systems and swap in and out
* Ability to actually interface with rules for different systems - ex. to enforce penalties/bonuses, etc. automatically

**Comment by dovrosenberg** [comment]
I'm thinking the next step here would be to create a hook whenever the weather changes that would pass the new weather state.  Then someone else could build a module for a particular system (or more generic) to watch for certain weather states and apply rules changes.

I have no sense of how many people are using this module, though, or whether anyone would care enough to create a module like that.  If you're interested, let me know and I'll add this capability.

**Comment by Issacar13** [comment]
You may have already noticed, but there is a module called "weather report" that handles extreme weather calculation. I am pleasantly surprised to have stumbled upon this thread and being able to use both Simple Weather and Weather Report seamlessly would be great. I don't know if weather and suches are important to my players, but I absolutely love being able this module.
